### PR TITLE
Piano: Add track Volume and improve QOL

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
This patch implements a couple of enhancements to the synthesizer engine:
- Each track has a volume control.
- The input and tooltips for all controls are improved.
- The noise channel is pitched, which allows for basic drum synthesis.

### CODE REVIEW

File: Userland/Applications/Piano/main.cpp [hunk 1]
```diff
@@ -104,6 +104,11 @@ int main(int argc, char** argv)
     });
     audio_thread->start();
 
+    auto main_widget_updator = Core::Timer::construct(150, [&] {
```
Comment (willmcpherson2) [inline Userland/Applications/Piano/main.cpp:107]: *updater
:P

File: Userland/Applications/Piano/RollWidget.cpp [hunk 2]
```diff
@@ -88,6 +89,63 @@ void RollWidget::paint_event(GUI::PaintEvent& event)
     int horizontal_notes_to_paint = horizontal_paint_area / m_note_width;
 
     GUI::Painter painter(*this);
+
+    // Draw the background, if necessary.
+    if (viewport_changed() || paint_area != m_background->height()) {
+        dbgln("background redraw");
```
Comment (willmcpherson2) [inline Userland/Applications/Piano/RollWidget.cpp]: This `dbgln` gets added and then removed in the next commit

File: Userland/Applications/Piano/RollWidget.h [hunk 3]
```diff
@@ -61,4 +62,10 @@ class RollWidget final : public GUI::ScrollableWidget {
     Optional<Gfx::IntPoint> m_note_drag_start;
     Optional<RollNote> m_note_drag_location;
     int m_drag_note;
+
+    RefPtr<Gfx::Bitmap> m_background;
+    int m_prev_zoom_level { m_zoom_level };
+    int m_prev_scroll_x { horizontal_scrollbar().value() };
+    int m_prev_scroll_y { vertical_scrollbar().value() };
+    bool viewport_changed();
```
Comment (willmcpherson2) [inline Userland/Applications/Piano/RollWidget.h:70]: `viewport_changed` can be `const`. It should also reside with the other methods.

**Comment by willmcpherson2** [comment]
1.

We write commit message titles in the imperative mood ("do this", "add that"), so the commit messages should be more like:

```
Userland: Piano: Optimize repaints
```

```
Userland: Piano: Add volume and QOL improvements
```

2.

Preferably, the Volume/QOL commit should be split up into 3 commits: the volume knob, the new knob scaling improvements and the rewrite of `Track::noise`. Or at the very least, the commit message should mention the `Track::noise` rewrite. Also, it's really not necessary to rename `max_attack` to `attack_max` etc.

Other than those two points, these changes seem reasonable. And while I don't really understand how the optimisation works, it seems sound (just repaint the necessary stuff).

File: Userland/Applications/Piano/RollWidget.cpp (hunk 2 above)
Comment (kleinesfilmroellchen) [reply Userland/Applications/Piano/RollWidget.cpp]: Yes, this is an issue with the previous commit that awaits merge. I think it's OK to leave it in if both PRs are merged successively.

File: Userland/Applications/Piano/main.cpp (hunk 1 above)
Comment (kleinesfilmroellchen) [reply Userland/Applications/Piano/main.cpp:107]: this belongs to the previous commit

**Comment by kleinesfilmroellchen** [comment]
2. I'm putting these into a single commit because they seem minor to me and are closely related. The Track::noise rewrite is part of the previous PR.

**Review by willmcpherson2 — APPROVED** [review]
This all looks good to me. Although try to keep your commits atomic. For example, renaming `max_attack` while also changing code that relies on `max_attack` makes the diff unnecessarily complicated. If your commit message has 3 dot points, meaning it does 3 things, it is usually reasonable to use 3 commits.

**Pull request merged**